src/request/src/network.js

```diff
@@ -387,52 +386,52 @@ export class KinveyRequest extends NetworkRequest {
         return response;
       })
       .catch((error) => {
-        if (error instanceof InvalidCredentialsError && retry === true) {
-          const activeUser = ActiveUserHelper.get(this.client);
+        if (retry && error instanceof InvalidCredentialsError) {
+          const activeUser = this.client.getActiveUser();
 
-          if (!isDefined(activeUser)) {
-            throw error;
-          }
-
-          const socialIdentities = isDefined(activeUser._socialIdentity) ? activeUser._socialIdentity : {};
-          const sessionKey = Object.keys(socialIdentities)
-            .find(sessionKey => socialIdentities[sessionKey].identity === SocialIdentity.MobileIdentityConnect);
-          const session = socialIdentities[sessionKey];
+          if (isDefined(activeUser)) {
+            const socialIdentity = isDefined(activeUser._socialIdentity) ? activeUser._socialIdentity : {};
+            const sessionKey = Object.keys(socialIdentity)
+              .find(sessionKey => socialIdentity[sessionKey].identity === 'kinveyAuth');
+            const oldSession = socialIdentity[sessionKey];
 
-          if (isDefined(session)) {
-            // Refresh MIC Token
-            if (session.identity === SocialIdentity.MobileIdentityConnect) {
-              const refreshMICRequest = new KinveyRequest({
+            if (isDefined(oldSession)) {
+              const request = new KinveyRequest({
                 method: RequestMethod.POST,
                 headers: {
                   'Content-Type': 'application/x-www-form-urlencoded'
                 },
                 authType: AuthType.App,
                 url: url.format({
-                  protocol: session.protocol || this.client.micProtocol,
-                  host: session.host || this.client.micHost,
+                  protocol: this.client.micProtocol,
```

Comment:
Similar to the previous question - why remove the check for `session.protocol` and `session.host`? Seems like it might miss some case.